python-sdk/src/astro/utils/table.py

```diff
@@ -1,97 +1,122 @@
+from __future__ import annotations
+
 import inspect
-from typing import Callable, Optional
+from typing import Callable
 
+from airflow.models.xcom_arg import XComArg
 from astro.sql.table import BaseTable
+from astro.utils.typing_compat import Context
 
 
-def _pull_first_table_from_parameters(parameters: dict) -> Optional[BaseTable]:
+def _have_same_conn_id(tables: list[BaseTable]) -> bool:
```

Comment:
`_have_same_conn_id` -> `_has_same_conn_id`

Reply:
I chose `have` because of `tables: list[BaseTable]` 😅